pkg/kubectl/sorting_printer_test.go

```diff
@@ -0,0 +1,166 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+func TestSortingPrinter(t *testing.T) {
+	tests := []struct {
+		obj   runtime.Object
+		sort  runtime.Object
+		field string
+	}{
+		{
+			obj: &api.PodList{
+				Items: []api.Pod{
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "a",
+						},
+					},
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "b",
+						},
+					},
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "c",
+						},
+					},
+				},
+			},
+			sort: &api.PodList{
+				Items: []api.Pod{
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "a",
+						},
+					},
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "b",
+						},
+					},
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "c",
+						},
+					},
+				},
+			},
+			field: "{.ObjectMeta.Name}",
+		},
+		{
+			obj: &api.PodList{
+				Items: []api.Pod{
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "b",
+						},
+					},
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "c",
+						},
+					},
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "a",
+						},
+					},
+				},
+			},
+			sort: &api.PodList{
+				Items: []api.Pod{
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "a",
+						},
+					},
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "b",
+						},
+					},
+					{
+						ObjectMeta: api.ObjectMeta{
+							Name: "c",
+						},
+					},
+				},
+			},
+			field: "{.ObjectMeta.Name}",
+		},
+		{
+			obj: &api.ReplicationControllerList{
+				Items: []api.ReplicationController{
+					{
+						Spec: api.ReplicationControllerSpec{
+							Replicas: 5,
+						},
+					},
+					{
+						Spec: api.ReplicationControllerSpec{
+							Replicas: 1,
+						},
+					},
+					{
+						Spec: api.ReplicationControllerSpec{
+							Replicas: 9,
+						},
+					},
+				},
+			},
+			sort: &api.ReplicationControllerList{
+				Items: []api.ReplicationController{
+					{
+						Spec: api.ReplicationControllerSpec{
+							Replicas: 1,
+						},
+					},
+					{
+						Spec: api.ReplicationControllerSpec{
+							Replicas: 5,
+						},
+					},
+					{
+						Spec: api.ReplicationControllerSpec{
+							Replicas: 9,
+						},
+					},
+				},
+			},
+			field: "{.Spec.Replicas}",
+		},
+	}
+	for _, test := range tests {
+		sort := &SortingPrinter{SortField: test.field}
+		if err := sort.sortObj(test.obj); err != nil {
+			t.Errorf("unexpected error: %v", err)
```

Comment:
Add a name field to the tests struct and pass it here like "%s: unexpected error" so we know which test fails if/when they fail. Also if you fail here you don't need to continue to `DeepEqual`.